parts/k8s/cloud-init/artifacts/cse_install.sh

```diff
@@ -177,6 +178,17 @@ downloadContainerd() {
     retrycmd_get_tarball 120 5 "$CONTAINERD_DOWNLOADS_DIR/${CONTAINERD_TGZ_TMP}" ${CONTAINERD_DOWNLOAD_URL} || exit $ERR_CONTAINERD_DOWNLOAD_TIMEOUT
 }
 
+installAPMZ() {
```

Comment:
are we going to pre-install this on the vhd? Also, does uploading metrics require outbound internet? if so, should be disable it/skip it in "locked down environments"?

Reply:
I'd like to pre-install. What do you think about that?

Reply:
actually we can probably add `{{- if HasTelemetryEnabled }}` here

Reply:
+1 for pre-installing

Reply:
But also keep the install path in there if customers are using other distros

Reply:
`apmz` is now pre-installed and will check to verify the version installed matches the version symlinked.